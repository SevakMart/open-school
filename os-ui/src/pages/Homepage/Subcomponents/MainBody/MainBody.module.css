.listContainer {
  position: relative;
}
.listContainer .icon {
  position: absolute;
  z-index: 2;
  top: 45%;
  font-size: 3rem;
  cursor: pointer;
}
.listContainer .icon__left {
  left: 0;
}
.listContainer .icon__right {
  right: 0;
}

.registrationButton {
  width: 40%;
  text-align: center;
}/*# sourceMappingURL=MainBody.module.css.map */