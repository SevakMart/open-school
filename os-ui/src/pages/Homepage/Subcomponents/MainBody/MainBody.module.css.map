{"version":3,"sources":["MainBody.module.scss","MainBody.module.css"],"names":[],"mappings":"AAIA;EACI,kBAAA;ACHJ;ADKI;EACE,kBAAA;EACA,UAAA;EACA,QAAA;EACA,eAAA;EACA,eAAA;ACHN;ADKM;EACE,OAAA;ACHR;ADKM;EACE,QAAA;ACHR;;ADOE;EACE,UAAA;EACA,kBAAA;ACJJ","file":"MainBody.module.css"}