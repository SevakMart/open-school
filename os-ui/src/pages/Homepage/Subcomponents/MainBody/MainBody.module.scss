@import "../../../../styles/Configs/Colors.scss";

  .registrationButton{
    text-align: center;
    margin: auto;
    width:40%;
  }
  
  .regButton {
    font-weight: 500;
    font-size: 16px;
    line-height: 20px;
    color: $header-main-color;
    background-color: $white-text-color;
    border: 1px solid $header-main-color;
    border-radius: 13px;
    padding: 16px 32px;
    cursor: pointer;
    &:hover {
      background-color: $secondary-color;
      color: $white-text-color;
      border: 1px solid transparent;
    }
  }
  
  .btnAbove {
    margin-bottom: 48px;
  }
  
  .btnBelow {
    margin-top: 48px;
    margin-bottom: 50px;
  }
  