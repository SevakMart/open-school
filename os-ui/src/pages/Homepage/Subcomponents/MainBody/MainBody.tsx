import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import Button from '../../../../component/Button/Button';
import LeftArrowIcon from '../../../../icons/LeftArrow';
import RightArrowIcon from '../../../../icons/RightArrow';
import { Types } from '../../../../types/types';
import { openModal } from '../../../../redux/Slices/PortalOpenStatus';
import styles from './MainBody.module.scss';

const MainBody = ({
  page, maxPage, children, clickPrevious, clickNext, isMentor,
}:
     /* eslint-disable-next-line max-len */
    {page:number, maxPage:number, children:any, clickPrevious:()=>void, clickNext:()=>void, isMentor:boolean}) => {
  const {
    listContainer, icon, icon__left, icon__right, registrationButton,
  } = styles;
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const handleSignUp = () => {
    dispatch(openModal(Types.Button.SIGN_UP));
  };

  return (
    <>
      <div className={listContainer} style={isMentor ? { margin: '6%' } : { backgroundColor: '#F9FAFD', margin: '0 6%' }}>
        {page > 0 && (
        <div className={`${icon} ${icon__left}`}>
          <LeftArrowIcon testId="categoryLeftArrow" handleArrowClick={() => { clickPrevious(); }} />
        </div>
        )}
        {children}
        {page < maxPage
        && (
        <div className={`${icon} ${icon__right}`}>
          <RightArrowIcon testId="categoryRightArrow" handleArrowClick={() => { clickNext(); }} />
        </div>
        )}
      </div>
      <div className={registrationButton} style={isMentor ? { margin: 'auto', padding: '5% 0 10%' } : { margin: '2rem auto 4rem' }}>
        <Button.SignUpButton className={['mainMentorRegistrationButton']} onClick={handleSignUp}>
          {t('button.homePage.registerMentor')}
        </Button.SignUpButton>
      </div>
    </>
  );
};
export default MainBody;
