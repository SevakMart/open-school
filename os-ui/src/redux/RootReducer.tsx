import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
