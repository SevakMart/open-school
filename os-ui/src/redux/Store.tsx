import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
