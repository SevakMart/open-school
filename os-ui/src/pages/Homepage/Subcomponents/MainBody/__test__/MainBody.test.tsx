import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../../../../redux/Store';
import MainBody from '../MainBody';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key:string) => key }),
}));

describe('Create test cases for MainBody component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Router>
        <Provider store={store}>
          <MainBody
            page={0}
            maxPage={10}
            children={<p>Hello world</p>}
            clickNext={jest.fn()}
            clickPrevious={jest.fn()}
            isMentor={false}
          />
        </Provider>
      </Router>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
