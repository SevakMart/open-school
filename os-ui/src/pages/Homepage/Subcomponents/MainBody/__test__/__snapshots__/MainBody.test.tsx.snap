// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for MainBody component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="listContainer"
    style="background-color: rgb(255, 255, 255); margin: 0px 3%;"
  >
    <p>
      Hello world
    </p>
  </div>
  <div
    class="registrationButton"
  >
    <a
      href="/categories/subcategories"
    >
      <button
        class="regButton btnBelow"
        type="button"
      >
        button.homePage.seeAll
      </button>
    </a>
  </div>
</DocumentFragment>
`;
